### PR TITLE
external controlplane setup with istioctl manifest install

### PR DESCRIPTION
**Pull request by ericvn**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Review by rvennam — APPROVED** [review]

**Comment by ericvn** [comment]
Test output has changed to:
```
This will install the Istio 1.10.0 remote profile with ["Istio core" "Istiod remote" "Ingress gateways"] components into the cluster. Proceed? (y/N) 
- Processing resources for Istio core.
✔ Istio core installed
- Processing resources for Ingress gateways, Istiod remote.
- Processing resources for Ingress gateways, Istiod remote. Waiting for Deployment/external-istio...
✔ Istiod remote installed
- Processing resources for Ingress gateways. Waiting for Deployment/external-istiod/istio-ingress...
✘ Ingress gateways encountered an error: failed to wait for resource: resources not ready after 5m0s: timed out waiting for the condition
Deployment/external-istiod/istio-ingressgateway
- Pruning removed resourcesError: failed to install manifests: errors occurred during operation
```
which we are ignoring by the earlier change.

Is there another change where the ignored failure is causing a problem in the external control plane which isn't being detected later in the test?

**Comment by ericvn** [comment]
If I back to look at the logs, before we moved from kubectl 1.15.0 to 1.20.0.  

We used `istioctl manifest generate | kubectl apply` and there were no error messages:
```
namespace/external-istiod created
customresourcedefinition.apiextensions.k8s.io/authorizationpolicies.security.istio.io created
customresourcedefinition.apiextensions.k8s.io/destinationrules.networking.istio.io created
customresourcedefinition.apiextensions.k8s.io/envoyfilters.networking.istio.io created
customresourcedefinition.apiextensions.k8s.io/gateways.networking.istio.io created
customresourcedefinition.apiextensions.k8s.io/istiooperators.install.istio.io created
customresourcedefinition.apiextensions.k8s.io/peerauthentications.security.istio.io created
customresourcedefinition.apiextensions.k8s.io/requestauthentications.security.istio.io created
customresourcedefinition.apiextensions.k8s.io/serviceentries.networking.istio.io created
customresourcedefinition.apiextensions.k8s.io/sidecars.networking.istio.io created
customresourcedefinition.apiextensions.k8s.io/virtualservices.networking.istio.io created
customresourcedefinition.apiextensions.k8s.io/workloadentries.networking.istio.io created
customresourcedefinition.apiextensions.k8s.io/workloadgroups.networking.istio.io created
serviceaccount/istio-ingressgateway-service-account created
serviceaccount/istio-reader-service-account created
serviceaccount/istiod-service-account created
clusterrole.rbac.authorization.k8s.io/istio-reader-external-istiod created
clusterrole.rbac.authorization.k8s.io/istiod-external-istiod created
clusterrolebinding.rbac.authorization.k8s.io/istio-reader-external-istiod created
clusterrolebinding.rbac.authorization.k8s.io/istiod-external-istiod created
validatingwebhookconfiguration.admissionregistration.k8s.io/istiod-external-istiod created
envoyfilter.networking.istio.io/metadata-exchange-1.10 created
envoyfilter.networking.istio.io/metadata-exchange-1.8 created
envoyfilter.networking.istio.io/metadata-exchange-1.9 created
envoyfilter.networking.istio.io/stats-filter-1.10 created
envoyfilter.networking.istio.io/stats-filter-1.8 created
envoyfilter.networking.istio.io/stats-filter-1.9 created
envoyfilter.networking.istio.io/tcp-metadata-exchange-1.10 created
envoyfilter.networking.istio.io/tcp-metadata-exchange-1.8 created
envoyfilter.networking.istio.io/tcp-metadata-exchange-1.9 created
envoyfilter.networking.istio.io/tcp-stats-filter-1.10 created
envoyfilter.networking.istio.io/tcp-stats-filter-1.8 created
envoyfilter.networking.istio.io/tcp-stats-filter-1.9 created
configmap/istio created
configmap/istio-sidecar-injector created
mutatingwebhookconfiguration.admissionregistration.k8s.io/istio-sidecar-injector-external-istiod created
deployment.apps/istio-ingressgateway created
poddisruptionbudget.policy/istio-ingressgateway created
role.rbac.authorization.k8s.io/istio-ingressgateway-sds created
role.rbac.authorization.k8s.io/istiod-external-istiod created
rolebinding.rbac.authorization.k8s.io/istio-ingressgateway-sds created
rolebinding.rbac.authorization.k8s.io/istiod-external-istiod created
horizontalpodautoscaler.autoscaling/istio-ingressgateway created
service/istio-ingressgateway created
serviceaccount/istiod-service-account created
secret/istio-kubeconfig created
This will install the Istio 1.10.0  profile with ["Istiod"] components into the cluster. Proceed? (y/N) 
```

Moving to kubectl 1.20.0 yields error messages and a returned error code (which is ignored in the other PR):
```
namespace/external-istiod created
customresourcedefinition.apiextensions.k8s.io/authorizationpolicies.security.istio.io created
customresourcedefinition.apiextensions.k8s.io/destinationrules.networking.istio.io created
customresourcedefinition.apiextensions.k8s.io/envoyfilters.networking.istio.io created
customresourcedefinition.apiextensions.k8s.io/gateways.networking.istio.io created
customresourcedefinition.apiextensions.k8s.io/istiooperators.install.istio.io created
customresourcedefinition.apiextensions.k8s.io/peerauthentications.security.istio.io created
customresourcedefinition.apiextensions.k8s.io/requestauthentications.security.istio.io created
customresourcedefinition.apiextensions.k8s.io/serviceentries.networking.istio.io created
customresourcedefinition.apiextensions.k8s.io/sidecars.networking.istio.io created
customresourcedefinition.apiextensions.k8s.io/virtualservices.networking.istio.io created
customresourcedefinition.apiextensions.k8s.io/workloadentries.networking.istio.io created
customresourcedefinition.apiextensions.k8s.io/workloadgroups.networking.istio.io created
serviceaccount/istio-ingressgateway-service-account created
serviceaccount/istio-reader-service-account created
serviceaccount/istiod-service-account created
clusterrole.rbac.authorization.k8s.io/istio-reader-external-istiod created
clusterrole.rbac.authorization.k8s.io/istiod-external-istiod created
clusterrolebinding.rbac.authorization.k8s.io/istio-reader-external-istiod created
clusterrolebinding.rbac.authorization.k8s.io/istiod-external-istiod created
validatingwebhookconfiguration.admissionregistration.k8s.io/istiod-external-istiod created
configmap/istio created
configmap/istio-sidecar-injector created
mutatingwebhookconfiguration.admissionregistration.k8s.io/istio-sidecar-injector-external-istiod created
deployment.apps/istio-ingressgateway created
poddisruptionbudget.policy/istio-ingressgateway created
role.rbac.authorization.k8s.io/istio-ingressgateway-sds created
role.rbac.authorization.k8s.io/istiod-external-istiod created
rolebinding.rbac.authorization.k8s.io/istio-ingressgateway-sds created
rolebinding.rbac.authorization.k8s.io/istiod-external-istiod created
horizontalpodautoscaler.autoscaling/istio-ingressgateway created
service/istio-ingressgateway created
unable to recognize "STDIN": no matches for kind "EnvoyFilter" in version "networking.istio.io/v1alpha3"
unable to recognize "STDIN": no matches for kind "EnvoyFilter" in version "networking.istio.io/v1alpha3"
unable to recognize "STDIN": no matches for kind "EnvoyFilter" in version "networking.istio.io/v1alpha3"
unable to recognize "STDIN": no matches for kind "EnvoyFilter" in version "networking.istio.io/v1alpha3"
unable to recognize "STDIN": no matches for kind "EnvoyFilter" in version "networking.istio.io/v1alpha3"
unable to recognize "STDIN": no matches for kind "EnvoyFilter" in version "networking.istio.io/v1alpha3"
unable to recognize "STDIN": no matches for kind "EnvoyFilter" in version "networking.istio.io/v1alpha3"
unable to recognize "STDIN": no matches for kind "EnvoyFilter" in version "networking.istio.io/v1alpha3"
unable to recognize "STDIN": no matches for kind "EnvoyFilter" in version "networking.istio.io/v1alpha3"
unable to recognize "STDIN": no matches for kind "EnvoyFilter" in version "networking.istio.io/v1alpha3"
unable to recognize "STDIN": no matches for kind "EnvoyFilter" in version "networking.istio.io/v1alpha3"
unable to recognize "STDIN": no matches for kind "EnvoyFilter" in version "networking.istio.io/v1alpha3"
```

and finally the above moving to `istioctl install`:
```
namespace/external-istiod created
This will install the Istio 1.10.0 remote profile with ["Istio core" "Istiod remote" "Ingress gateways"] components into the cluster. Proceed? (y/N) 
- Processing resources for Istio core.
✔ Istio core installed
- Processing resources for Ingress gateways.
- Processing resources for Ingress gateways, Istiod remote.
- Processing resources for Ingress gateways, Istiod remote. Waiting for Deployment/external-istio...
✔ Istiod remote installed
- Processing resources for Ingress gateways. Waiting for Deployment/external-istiod/istio-ingress...
✘ Ingress gateways encountered an error: failed to wait for resource: resources not ready after 5m0s: timed out waiting for the condition
Deployment/external-istiod/istio-ingressgateway
- Pruning removed resourcesError: failed to install manifests: errors occurred during operation
```

I'm still not sure the external control plane is woking as desired. Does the test really test the remote control plane's functionality.

**Comment by ericvn** [comment]
Continuing through  the logs for using kubectl 1.2.0, we do see that the ingress-gateway pod goes to Running and that traffic is being sent, so everything passes.

With the istioctl install, we fail to see the ingress-gateway pod start within the 5 mins (which we ignore), but we verify it later as part of the test and the pod basically takes 6:30 minutes to get to Running...

File: content/en/docs/setup/additional-setup/external-controlplane/index.md [hunk 1]
```diff
@@ -247,7 +247,7 @@ Install the configuration on the remote cluster:
 
 {{< text bash >}}
 $ kubectl create namespace external-istiod --context="${CTX_REMOTE_CLUSTER}"
-$ istioctl manifest generate -f remote-config-cluster.yaml  | kubectl apply --context="${CTX_REMOTE_CLUSTER}" -f -
+$ istioctl manifest install -f remote-config-cluster.yaml --context="${CTX_REMOTE_CLUSTER}"
```
Comment (frankbu) [inline content/en/docs/setup/additional-setup/external-controlplane/index.md:250]: In 1.9 `istioctl install` will not work in the remote cluster (at least not in a real, non-test, env). That's why the doc explicitly uses `istio manifest generate | kubectl apply`. Has that been fixed recently, where it's now supposed to work?

Also, why do we seem to have two istioctl commands that do the same thing (`istioctl install` and `istioctl manifest install`)?

Comment (ericvn) [reply content/en/docs/setup/additional-setup/external-controlplane/index.md:250]: Maybe I should have left the `manifest` out? And thanks for the input on `install` not working for 1.9. That would pretty much mean that I can cancel this PR.

**Comment by ericvn** [comment]
I had submitted the `doc.test.multicluster_istio.io` test again to see if the ingress gateway took nearly 6:30 minutes to become running, and it did.

**Comment by ericvn** [comment]
Based on Frank's comment that `istioctl install` won't work in real life, and that the ingress-gateway takes a long time to become running, It makes sense to close this PR.

**Pull request closed without merging**